### ISSUE TITLE
Undefined property: stdClass::$image in 1.8.2

### ISSUE THREAD

**Issue by superDuperCyberTechno** [getgrav/grav-plugin-admin#1454]
I get the error `Undefined property: stdClass::$image` when uploading an image to a page template. The error handler specifically references [line 722 in adminbasecontroller.php](https://github.com/getgrav/grav-plugin-admin/blob/develop/classes/adminbasecontroller.php#L722). The image specific blueprint is here:

```
            header.image:
                type: file
                label: Screenshot
                limit: 1
                destination: 'self@'
                multiple: false
                style: vertical
                accept:
                  - image/*
```

The image gets uploaded regardless of error, but the header is not updated...

Rolling back to 1.8.1 fixed it for now.

**Comment by rhukster** [comment]
Don't do this, pagemedia field already uploads to the page folder.  File field is intended to be used in other non-page blueprints.

**Comment by superDuperCyberTechno** [comment]
So, what would the correct way be to achieve this, blueprint wise?

**Comment by herrbischoff** [comment]
Same here. Since the update to Grav 1.4.5 and Admin 1.8.2, I cannot upload images, resulting in the same error @superDuperCyberTechno reported.

My blueprint excerpt is a follows:

```
header.images:
  type: file
  label: Images
  random_name: true
  multiple: true
  limit: 2
  destination: 'assets'
  filesize: 10
  accept:
    - image/*
```

This has worked thus far without issue. The update broke it.

If there are breaking changes, it would be advantageous to know about them **before** the update. If there are best practices, it would be great to know about the **at all** instead of just getting a response like

> Don't do this, pagemedia field already uploads to the page folder. File field is intended to be used in other non-page blueprints.

without any pointers as to why. I created the blueprint according to the documentation at the time and it worked. Either this is a bug or a breaking, non-documented change.

**Comment by rhukster** [comment]
We didn’t realize this was a breaking change because this field really is not intended to replace pagemedia picker. Now we have a couple of reports of people using filepicker in conjunction with pagemedia picker, we’ll investigate and find a solution. Sorry about this!

**Comment by rhukster** [comment]
FYI the recommended approach is to use the standard pagemedia field to upload images into a page and a filepicker to select them for specific header variables

**Comment by herrbischoff** [comment]
> FYI the recommended approach is to use the standard pagemedia field to upload images into a page and a filepicker to select them for specific header variables

This recommended approach strikes me as counter-intuitive. Uploading and assigning a file to a field should be one action, not two. Teaching users to do it in two steps would be frustrating and should not be necessary.

In any case, thanks for looking into it.

**Comment by mahagr** [comment]
Issue can be fixed by changing line 722 in adminbasecontroller.php to have `isset()`:

```
if (isset($obj->header()->{$init_key})) {
```

**Comment by mahagr** [comment]
This issue has been fixed in Admin 1.8.3
